rasa/server.py

```diff
@@ -0,0 +1,740 @@
+        """Evaluate intents against a Rasa model."""
+        validate_request_body(
+            request,
+            "You must provide some nlu data in the request body in order to "
+            "evaluate your model.",
+        )
+
+        nlu_data = rasa.utils.io.create_temporary_file(request.body, mode="w+b")
+        data_path = os.path.abspath(nlu_data)
+
+        if not os.path.exists(app.agent.model_directory):
+            raise ErrorResponse(409, "Conflict", "Loaded model file not found.")
+
+        model_directory = app.agent.model_directory
+        _, nlu_model = get_model_subdirectories(model_directory)
+
+        try:
+            evaluation = run_evaluation(data_path, nlu_model)
+            return response.json(evaluation)
+        except Exception as e:
+            logger.debug(traceback.format_exc())
+            raise ErrorResponse(
+                500,
+                "ServerError",
+                "An unexpected error occurred during evaluation. Error: {}".format(e),
+            )
+
+    @app.post("/model/predict")
+    @requires_auth(app, auth_token)
+    @ensure_loaded_agent(app)
+    async def tracker_predict(request: Request):
+        """ Given a list of events, predicts the next action"""
+        validate_request_body(
+            request,
+            "No events defined in request_body. Add events to request body in order to "
+            "predict the next action.",
+        )
+
+        sender_id = UserMessage.DEFAULT_SENDER_ID
+        verbosity = event_verbosity_parameter(request, EventVerbosity.AFTER_RESTART)
+        request_params = request.json
+
+        try:
+            tracker = DialogueStateTracker.from_dict(
+                sender_id, request_params, app.agent.domain.slots
+            )
+        except Exception as e:
+            logger.debug(traceback.format_exc())
+            raise ErrorResponse(
+                400,
+                "BadRequest",
+                "Supplied events are not valid. {}".format(e),
+                {"parameter": "", "in": "body"},
+            )
+
+        try:
+            policy_ensemble = app.agent.policy_ensemble
+            probabilities, policy = policy_ensemble.probabilities_using_best_policy(
+                tracker, app.agent.domain
+            )
+
+            scores = [
+                {"action": a, "score": p}
+                for a, p in zip(app.agent.domain.action_names, probabilities)
+            ]
+
+            return response.json(
+                {
+                    "scores": scores,
+                    "policy": policy,
+                    "tracker": tracker.current_state(verbosity),
+                }
+            )
+        except Exception as e:
+            logger.debug(traceback.format_exc())
+            raise ErrorResponse(
+                500, "ServerError", "An unexpected error occurred. Error: {}".format(e)
+            )
+
+    @app.post("/model/parse")
+    @requires_auth(app, auth_token)
```

Comment:
needs `@ensure_loaded_agent(app)`

Reply:
`@ensure_loaded_agent(app)` is not required in this case.
We just need `agent.interpreter`. If just an NLU model is provided, the interpreter is set. If just a core model is provided, the interpreter is set to `RegexInterpreter`. So this endpoint should work in any case.

Reply:
I see, makes sense :+1: